src/Avalonia.Base/Input/PointerOverPreProcessor.cs

```diff
@@ -41,14 +41,16 @@ public void OnNext(RawInputEventArgs value)
                     _lastActivePointerDevice = pointerDevice;
                 }
 
-                if (args.Type is RawPointerEventType.LeaveWindow or RawPointerEventType.NonClientLeftButtonDown 
-                    or RawPointerEventType.TouchCancel or RawPointerEventType.TouchEnd
-                    && _currentPointer is var (lastPointer, lastPosition))
+                if (args.Type is RawPointerEventType.LeaveWindow or RawPointerEventType.NonClientLeftButtonDown
+                    or RawPointerEventType.TouchCancel or RawPointerEventType.TouchEnd)
                 {
-                    _currentPointer = null;
-                    ClearPointerOver(lastPointer, args.Root, 0, PointToClient(args.Root, lastPosition),
-                        new PointerPointProperties(args.InputModifiers, args.Type.ToUpdateKind()),
-                        args.InputModifiers.ToKeyModifiers());
+                    if (_currentPointer is var (lastPointer, lastPosition))
+                    {
```

Comment:
we don't want to clear PointerOver (since there is none) - but we also do not want to set any later in this conditions chain.